FeedKit/images/ImageRepository.swift

```diff
@@ -325,28 +262,33 @@ extension ImageRepository: Images {
       contentModes: nil
     )
   }
-
+  
+  private static func makeProcessors(id: FKImage.ID) -> [ImageProcessing] {
+    guard !id.isClean else {
+      return []
+    }
+    
+    return [
+      ImageProcessor.Resize(size: id.size, upscale: true), 
```

Comment:
It might be less efficient than your custom implementation because it draws twice. There should be a better way to make the processors composable without compromising on performance. I'm thinking something like:

```swift
let processor = ImageProcessor.CoreGraphics()
    .resize(to: id.size, crop: true, upscale: true)
    .addRoundedCorners(radius: 3)
```

Reply:
I couldn't find a design which I truly liked in any of the existing frameworks. They either end up having a type explosion or cramming everything into a single processor.

Reply:
I’m motivating myself to contribute the stuff I need from Nuke. 😋 Had to remove the outline for Dark Mode, will use a custom view with backdrop. A composite processor for common things sounds good. On the other hand, resize with rounded corners could be merged into one, with a default radius of 0. I mean, this use case is super standard.

Reply:
Sidebar, IB is the least intrusive way of implementing Dark Mode.

Reply:
Hmm, yeah, I might be overcomplicating things. It could be simply `ImageProcessor.Resize(size:upscale:...)` for resizing, and `ImageProcessor.CornerRadius(size:radius:)` for corner radius. The presence of the `size` parameter would make it clear that you _must_ ensure that the images are resized to match the view size. And it would also upscale the images by default, removing the burden of thinking about these details from the user.